### PR TITLE
Use daemon to get passphrase

### PR DESCRIPTION
**Pull request by patrickxb**
Signup, change passphrase, recover passphrase all used a secret ui on the client and were unable to use delegated secret ui.  This PR fixes that (issue CORE-2205).

r? @maxtaco

### CODE REVIEW

File: go/client/cmd_passphrase_change.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ func NewCmdPassphraseChange(cl *libcmdline.CommandLine) cli.Command {
 }
 
 func (c *CmdPassphraseChange) Run() error {
-	pp, err := promptNewPassphrase()
+	pp, err := PromptNewPassphrase(G)
```
Comment (maxtaco) [inline go/client/cmd_passphrase_change.go]: can you contextify this command?

File: go/client/cmd_passphrase_recover.go [hunk 2]
```diff
@@ -31,7 +31,7 @@ func (c *CmdPassphraseRecover) Run() error {
 	if err := c.confirm(); err != nil {
 		return err
 	}
-	pp, err := promptNewPassphrase()
+	pp, err := PromptNewPassphrase(G)
```
Comment (maxtaco) [inline go/client/cmd_passphrase_recover.go]: sam here

File: go/client/cmd_signup.go [hunk 3]
```diff
@@ -460,3 +454,52 @@ func (s *CmdSignup) handlePostError(inerr error) (retry bool, err error) {
...
+	arg.Prompt = "Please reenter your passphrase for confirmation"
+	_, err = s.promptPassphraseUntilCheck(arg, match)
+	if err != nil {
+		return keybase1.GetPassphraseRes{}, err
+	}
+
+	return res, nil
+}
+
+func (s *CmdSignup) promptPassphraseUntilCheck(arg keybase1.GUIEntryArg, checker *libkb.Checker) (keybase1.GetPassphraseRes, error) {
```
Comment (maxtaco) [inline go/client/cmd_signup.go]: do you need this code in both here and `client/util`?

**Comment by maxtaco** [comment]
Some housekeeping and larger design question...

**Comment by patrickxb** [comment]
PTAL

File: go/client/cmd_passphrase_change.go [hunk 4]
```diff
@@ -9,25 +9,27 @@ import (
...
 
 func (c *CmdPassphraseChange) Run() error {
-	pp, err := promptNewPassphrase()
+	pp, err := PromptNewPassphrase(c.G())
 	if err != nil {
 		return err
 	}
 
-	if err := passphraseChange(G, newChangeArg(pp, false)); err != nil {
+	if err := passphraseChange(c.G(), newChangeArg(pp, false)); err != nil {
```
Comment (maxtaco) [inline go/client/cmd_passphrase_change.go]: now that we're contextified, can you nuke `GlobUI` below?

**Comment by maxtaco** [comment]
I get a panic in standalone mode:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x28 pc=0x42e9012]

goroutine 42 [running]:
github.com/keybase/client/go/service.(*AccountHandler).PassphrasePrompt(0xc820217aa0, 0x629e5f8, 0xc8201f0b40, 0x0, 0xc82022f1f0, 0xa, 0xc820229050, 0x29, 0xc82022f1d0, 0x6, ...)
	/Users/max/src/go/src/github.com/keybase/client/go/service/account.go:35 +0xa2
github.com/keybase/client/go/protocol.AccountProtocol.func4(0x629e5f8, 0xc8201f0b40, 0x48f6180, 0xc8201eecc0, 0x0, 0x0, 0x0, 0x0)
	/Users/max/src/go/src/github.com/keybase/client/go/protocol/keybase_v1.go:86 +0x14e
github.com/keybase/client/go/vendor/github.com/keybase/go-framed-msgpack-rpc.(*callRequest).Serve.func1(0xc82024ebd0, 0x62606c8, 0xc8201d3ad0, 0x0, 0x0, 0x48f6180, 0xc8201eecc0, 0x4fbebf0, 0xc8201eeca0, 0x629e250, ...)
	/Users/max/src/go/src/github.com/keybase/client/go/vendor/github.com/keybase/go-framed-msgpack-rpc/request.go:108 +0x1ef
created by github.com/keybase/client/go/vendor/github.com/keybase/go-framed-msgpack-rpc.(*callRequest).Serve
	/Users/max/src/go/src/github.com/keybase/client/go/vendor/github.com/keybase/go-framed-msgpack-rpc/request.go:115 +0x270


```

**Comment by maxtaco** [comment]
The behavior in daemon mode without electron running is as before, is this correct?  Just checking...

**Comment by maxtaco** [comment]
When falling back to terminal mode, I get weird behavior too:

```
Your desired username: max39992
A public name for this device [home computer]: 
Please enter the passphrase to unlock your secret key: 
Please enter the passphrase to unlock your secret key: 
```

That's not the prompt I would expect, and also, nothing I can do advances the experience here.

**Comment by maxtaco** [comment]
(Confirmed we get the right behavior in master here)

File: go/client/util.go [hunk 5]
```diff
@@ -4,14 +4,76 @@
 package client
 
 import (
+	"golang.org/x/net/context"
```
Comment (maxtaco) [inline go/client/util.go]: can you rename this file away from `util.go` and to something that indicates that it's mainly concerned with passphrase?

File: go/client/cmd_signup.go [hunk 6]
```diff
@@ -460,3 +454,52 @@ func (s *CmdSignup) handlePostError(inerr error) (retry bool, err error) {
 
 	return
 }
+
+func (s *CmdSignup) promptPassphrase() (keybase1.GetPassphraseRes, error) {
```
Comment (maxtaco) [inline go/client/cmd_signup.go]: I couldn't trace it through.  Who calls this function?

**Comment by patrickxb** [comment]
I can't reproduce your terminal mode comments.  `signup` using daemon or standalone gives this:

    Your desired username: kwkwkw
    A public name for this device [home computer]: 
    Pick a strong passphrase (12+ characters): 
    Please reenter your passphrase for confirmation:

**Comment by maxtaco** [comment]
I did this on Linux, does that make a difference one way or the other?

**Comment by patrickxb** [comment]
I don't see how it would.  I did terminal tests with KEYBASE_PINENTRY=0.

**Comment by maxtaco** [comment]
(And I SSh'ed into the machine, which also might make a difference)

**Comment by patrickxb** [comment]
1. panic is fixed
2. yes, without electron in daemon mode will use gpg pinentry if available, just like it used to

**Comment by patrickxb** [comment]
Are you sure you were running this branch on the machine you ssh'ed into?

I just ssh'ed into a linux machine, built this branch, and tried it.  Same result:

patrick@xbdev:~$ ~/bin/keybase --standalone signup
▶ WARNING Running in devel mode
Your email address: jskdfjs@jkwejf.com
Your invite code (leave blank if you don't have one): nytimes
Your desired username: jekrjekrj
A public name for this device [home computer]: 
Pick a strong passphrase (12+ characters): 
Please reenter your passphrase for confirmation:

**Comment by maxtaco** [comment]
Hmmm, you're right, I can't repro.  I wonder if I was on master?

File: go/client/util.go (hunk 5 above)
Comment (patrickxb) [inline go/client/util.go]: done.

File: go/client/cmd_signup.go (hunk 6 above)
Comment (patrickxb) [inline go/client/cmd_signup.go]: removed.

**Comment by maxtaco** [comment]
Maybe my config was screwy.  I can't repro that issue now.  Sorry about that.

File: go/client/cmd_passphrase_change.go (hunk 4 above)
Comment (patrickxb) [inline go/client/cmd_passphrase_change.go]: done

**Comment by patrickxb** [comment]
So is this good to go?

**Comment by maxtaco** [comment]
It looks great!

**Pull request merged**